src/magnet_uri.cpp

```diff
@@ -230,6 +230,33 @@ namespace libtorrent
 			return;
 		}
 
+		std::string::size_type peer_pos = std::string::npos;
+		std::string peer = url_has_argument(uri, "x.pe", &peer_pos);
+		while (!peer.empty())
+		{
+			std::string::size_type divider = peer.find_last_of(':');
```

Comment:
I'm pretty sure the logic to parse an IPv4/IPv6 endpoint exists somewhere else in libtorrent. let me dig it up